### PR TITLE
feat(mojaloop/#2881): add bulk transfer timeout tests

### PR DESCRIPTION
**Pull request by kleyow**
feat(mojaloop/#2881): add bulk transfer timeout tests - [https://github.com/mojaloop/project/issues/2881](https://github.com/mojaloop/project/issues/2881)

- Added test for scenario - POST /bulkTransfers prepare request which has timed out. Assert both Payer and Payee receive PUT /bulkTransfers/{ID} callbacks detailing a COMPLETED bulkTransfer with individualTransfer errors for all transfers.

- Added test for scenario  - POST /bulkTransfers prepare request which has timed out and the Payee sends a late PUT /bulkTransfers/{ID} fulfilment request. Assert both Payer and Payee receive PUT /bulkTransfers/{ID} callbacks detailing a COMPLETED bulkTransfer with individualTransfer errors for all transfers. Positions don't change after late fulfilment.

### CODE REVIEW

File: collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json [hunk 1]
```diff
@@ -0,0 +1,838 @@
...
+              "exec": [
+                ""
+              ]
+            }
+          }
+        },
+        {
+          "id": 7,
+          "meta": {
+            "info": "Check Payerfsp position after Abort"
```
Comment (elnyry-sam-k) [inline collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json:282]:
```suggestion
            "info": "Check Payerfsp position after timeout"
```

File: collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json [hunk 2]
```diff
@@ -0,0 +1,838 @@
...
+              {
+                "id": 1,
+                "description": "Status code is 200",
+                "exec": [
+                  "expect(response.status).to.equal(200)"
+                ]
+              },
+              {
+                "id": 2,
+                "description": "Payerfsp position after Payer ABORT should be same as position before prepare.",
```
Comment (elnyry-sam-k) [inline collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json:309]:
```suggestion
                "description": "Payerfsp position after (bulk) timeout should be same as position before prepare.",
```

File: collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json [hunk 3]
```diff
@@ -0,0 +1,838 @@
...
+                  "expect(response.body[0].value).to.equal(+environment.payerfspPositionBeforePrepare)"
+                ]
+              }
+            ]
+          }
+        },
+        {
+          "id": 8,
+          "meta": {
+            "info": "Check Payeefsp position after Abort"
```
Comment (elnyry-sam-k) [inline collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json:320]:
```suggestion
            "info": "Check Payeefsp position after timeout"
```

File: collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json [hunk 4]
```diff
@@ -0,0 +1,838 @@
...
+              {
+                "id": 1,
+                "description": "Status code is 200",
+                "exec": [
+                  "expect(response.status).to.equal(200)"
+                ]
+              },
+              {
+                "id": 2,
+                "description": "Payeefsp position after Payee ABORT should be same as position before prepare.",
```
Comment (elnyry-sam-k) [inline collections/hub/other_tests/bulk_transfers/bulk-transfer-timeout-scenario.json:346]:
```suggestion
                "description": "Payeefsp position after timeout should be same as position before prepare.",
```

**Comment by elnyry-sam-k** [comment]
Thanks Kevin @kleyow , looking good. just minor description changes..

Separately we need to discuss the behavior for the second scenario and create a follow-up issue for that.. : what is the expected behavior - should we log an error, should we send an error callback or patch notification or similar..

**Comment by kleyow** [comment]
@elnyry-sam-k done

**Review by elnyry-sam-k — APPROVED** [review]

**Pull request merged**